pkg/util/pod/pod_test.go

```diff
@@ -0,0 +1,206 @@
+/*
+CCopyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func TestHasGate(t *testing.T) {
+	testCases := map[string]struct {
+		gateName string
+		pod      corev1.Pod
+		want     bool
+	}{
+		"scheduling gate present": {
+			gateName: "example.com/gate",
+			pod: corev1.Pod{
+				Spec: corev1.PodSpec{
+					SchedulingGates: []corev1.PodSchedulingGate{
+						{
+							Name: "example.com/gate",
+						},
+					},
+				},
+			},
+			want: true,
+		},
+		"another gate present": {
+			gateName: "example.com/gate",
+			pod: corev1.Pod{
+				Spec: corev1.PodSpec{
+					SchedulingGates: []corev1.PodSchedulingGate{
+						{
+							Name: "example.com/gate2",
+						},
+					},
+				},
+			},
+			want: false,
+		},
+		"no scheduling gates": {
+			pod: corev1.Pod{
+				Spec: corev1.PodSpec{
+					SchedulingGates: []corev1.PodSchedulingGate{},
```

Comment:
```suggestion
```

What if we remove this field?

Reply:
Dropped, it passes when using `cmpopts.EquateEmpty()`, otherwise because nil != empty.